src/NuGet.Core/NuGet.Build.Tasks/NuGet.RestoreEx.targets

```diff
@@ -28,6 +28,7 @@ Copyright (c) .NET Foundation. All rights reserved.
         NoCache="$(RestoreNoCache)"
         NoHttpCache="$(RestoreNoHttpCache)"
         ProjectFullPath="$(MSBuildProjectFullPath)"
```

Comment:
I’m still trying to understand static graph restore functionality, but I am curious to understand why NuGet needs to know that a project file is *virtual*. From my perspective, “virtual project” support feels like an SDK feature.

Could we instead:

- Generate the virtual project file from the text file in the .NET SDK layer.
- Pass its path to NuGet via `ProjectFullPath`, exactly the same way we do for an actual project.
- Keep the NuGet layer unaware of whether the file is real or synthetic.

That approach seems like it would:

- Remove the extra NuGet–SDK coupling.  
- Let the SDK fully own the abstraction.
- Keep NuGet focused on restore logic.

Is there any specific reason that makes it necessary for NuGet to know the file is virtual, rather than just receiving a path?